test/e2e/reconciliation_test.go

```diff
@@ -130,111 +131,114 @@ var _ = Describe("Reconciliation From OLSConfig CR", Ordered, func() {
 
 	It("should reconcile app configmap after changing application settings", func() {
 		By("fetch the app deployment generation")
-		deployment := &appsv1.Deployment{
-			ObjectMeta: metav1.ObjectMeta{
-				Name:      AppServerDeploymentName,
-				Namespace: OLSNameSpace,
-			},
-		}
-		err = client.Get(deployment)
-		Expect(err).NotTo(HaveOccurred())
-		generation := deployment.Generation
-
-		By("update LogLevel in the OLSConfig CR")
-		err = client.Get(cr)
-		Expect(err).NotTo(HaveOccurred())
-		if cr.Spec.OLSConfig.LogLevel == "DEBUG" {
-			cr.Spec.OLSConfig.LogLevel = "INFO"
-		} else {
-			cr.Spec.OLSConfig.LogLevel = "DEBUG"
-		}
-		err = client.Update(cr)
-		Expect(err).NotTo(HaveOccurred())
-		configMap := &corev1.ConfigMap{
-			ObjectMeta: metav1.ObjectMeta{
-				Name:      AppServerConfigMapName,
-				Namespace: OLSNameSpace,
-			},
-		}
-
-		By("wait for the app configmap to be updated")
-		err = client.WaitForConfigMapContainString(configMap, AppServerConfigMapKey, "app_log_level: "+cr.Spec.OLSConfig.LogLevel)
-		Expect(err).NotTo(HaveOccurred())
+		err := retry.RetryOnConflict(retry.DefaultRetry, func() error {
+			deployment := &appsv1.Deployment{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      AppServerDeploymentName,
+					Namespace: OLSNameSpace,
+				},
+			}
+			err = client.Get(deployment)
+			Expect(err).NotTo(HaveOccurred())
+			generation := deployment.Generation
 
-		By("check the app deployment generation that should be inscreased")
-		err = client.WaitForDeploymentCondition(deployment, func(dep *appsv1.Deployment) (bool, error) {
-			if dep.Generation <= generation {
-				return false, fmt.Errorf("current generation %d is inferior to observed generation %d", dep.Generation, generation)
+			By("update LogLevel in the OLSConfig CR")
+			err = client.Get(cr)
+			Expect(err).NotTo(HaveOccurred())
+			if cr.Spec.OLSConfig.LogLevel == "DEBUG" {
+				cr.Spec.OLSConfig.LogLevel = "INFO"
+			} else {
+				cr.Spec.OLSConfig.LogLevel = "DEBUG"
+			}
+			err = client.Update(cr)
```

Comment:
using `retry.RetryOnConflict`, we should return the error from such an Update operation and put the `Expect(err).NotTo(HaveOccurred())` outside the `retry.RetryOnConflict` to check if it fails after all these retries.